sdk/python/kfp/dsl/_metadata.py

```diff
@@ -48,10 +48,15 @@ def to_dict_or_str(self):
   @staticmethod
   def from_dict_or_str(json):
     type_meta = TypeMeta()
+    if isinstance(json, str) and '{' in json:
```

Comment:
It's kind of hacky. Could you explain why this is needed?

Reply:
This is to detect whether it is a str or a serialized str of a dict. Does that make sense?

Reply:
Let me separate the logic into a function called de-serialize

Reply:
Updated to remove the hacking way to detect the serialized string in the new PR.